### DISCUSSION TITLE
[macOS] macOS build doesn't support Entity Framework

### DISCUSSION THREAD

**Discussion by bstrother1** [unoplatform/uno#8192] [Q&A]
### Current behavior

MacOS build is stuck on Net Framework 2 which is so old that Entity Framework does not support it even at its earliest NuGet stable build https://www.nuget.org/packages/EntityFramework/4.2.0 which requires minimum Net Framework 4.0 nor does Net Framework Core which earliest build requires minimum 4.5.1. 

### Expected behavior

Expected macOS build to support Entity Framework usage like UWP and WPF builds do.

### How to reproduce it (as minimally and precisely as possible)

Follow https://docs.microsoft.com/en-us/ef/core/get-started/overview/first-app?tabs=netcore-cli

[MyApp01.zip](https://github.com/unoplatform/uno/files/8142266/MyApp01.zip)

### Workaround

None

### Works on UWP/WinUI

Yes

### NuGet package version(s)

   PackageReference Include="Uno.UI" Version="4.0.13" 
   PackageReference Include="Uno.UI.RemoteControl" Version="4.0.13" Condition="'$(Configuration)'=='Debug'" 
   PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.0.13" 
   PackageReference Include="Microsoft.Extensions.Logging" Version="5.0.0" 
   PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0" 
   PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="6.0.0" 
   PackageReference Include="Microsoft.Data.Sqlite.Core" Version="5.0.14" 
   PackageReference Include="Microsoft.EntityFrameworkCore" Version="5.0.14" 
   PackageReference Include="Microsoft.EntityFrameworkCore.Analyzers" Version="5.0.14" 
   PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="5.0.14"
   PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="5.0.14" 
   PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="5.0.14" 
   PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="5.0.14"
   PackageReference Include="System.Data.SQLite.Core"

### Affected platforms

macOS

### IDE

Visual Studio 2019

### IDE version

8.10

**Comment by jeromelaban** [comment]
Thanks for the question. Xamarin.mac is indeed supporting `netstandard2.1` as a maximum. 

If you want to run packages above this version, you'll need to go with the `net6.0-mac` projects from net6 and use [Entity Framework Core](https://docs.microsoft.com/en-us/ef/core/). These net6 targets are still in preview at this point, because .NET is currently still in preview (Preview 13, currently).

**Reply by bstrother1** [reply]
Why then if it can go up to netstandard2.1 am I getting "Load failed: Object reference not set to an instance of an object" if I change <TargetFrameworkVersion>v2.0</TargetFrameworkVersion> to <TargetFramework>netstandard2.1</TargetFramework>?

**Reply by bstrother1** [reply]
Tag changed from TargetFrameworkVersion to TargetFramework. For some reason the github commenting system removed my tags.

**Reply by bstrother1** [reply]
Upon changing the .csproj to netstandard2.1 closing and reopening the project a message appears stating Project does not support framework 'Xamarin.Mac, Version=v4.0'. Is this a Uno bug?

**Reply by bstrother1** [reply]
If someone wants to jump in and help it would be appreciated.

**Reply by MartinZikmund** [reply]
@bstrother1 Can you send a sample repo (zip or Github would be enough)? I would try locally. Also - did you also try using Rider on mac?

**Reply by MartinZikmund** [reply]
I have been able to reference `netstandard2.1` class library from `macOS` project without issues locally, so the repro would help

**Reply by bstrother1** [reply]
[MyApp01.zip](https://github.com/unoplatform/uno/files/8142266/MyApp01.zip)

I believe the attached example demonstrates what I mean. If I go into the .csproj in the mac head of this project and edit the target to netstandard2.1 it just errors out the project. No, I haven't attempted Rider.

**Reply by bstrother1** [reply]
> I have been able to reference `netstandard2.1` class library from `macOS` project without issues locally, so the repro would help

Could you attach a sample project of you doing this so I can see if it is my local build failing?

**Reply by bstrother1** [reply]
Just to update this discussion https://docs.microsoft.com/en-us/visualstudio/releasenotes/vs2019-mac-relnotes under Visual Studio 2019 for Mac version 8.10.15 (8.10.15.32), Old .NET x64 SDKs are not supported on Arm64 and must be removed before installing the latest supported versions. So the old template (non-net6) cannot be upgraded to netstandard2.0 and is essentially stranded on xamarin 2.0. I hope this saves someone else time.

**Reply by ajpinedam** [reply]
Thanks @bstrother1  for bringing this up!
